chain/client/src/client.rs

```diff
@@ -2382,14 +2403,33 @@ impl Client {
                             true,
                         ),
                         shards_to_split,
-                        BlocksCatchUpState::new(sync_hash, epoch_id),
+                        BlocksCatchUpState::new(sync_hash, epoch_id.clone()),
                     )
                 });
 
             // For colour decorators to work, they need to printed directly. Otherwise the decorators get escaped, garble output and don't add colours.
             debug!(target: "catchup", ?me, ?sync_hash, progress_per_shard = ?format_shard_sync_phase_per_shard(&shards_to_split, false), "Catchup");
             let use_colour = matches!(self.config.log_summary_style, LogSummaryStyle::Colored);
 
+            let tracking_shards: Vec<u64> =
```

Comment:
`run_catchup()` is triggered every `catchup_step_period`.
if I'm reading the code right, you are sending `StartSync` requests for every shard every invocation of the function. Is that correct? If yes, isn't that too much?